### PR TITLE
feat: Update Cohere integration with new default model, examples, minor docs fixes, and a new README

### PR DESCRIPTION
**Pull request by vblagoje**
Update:
- default model, use `command-r` for larger context window
- new examples
- new Readme for better first contact on pypi and beyond

### CODE REVIEW

**Review by shadeMe — APPROVED** [review]

File: integrations/cohere/README.md [hunk 1]
```diff
@@ -3,65 +3,219 @@
 [![PyPI - Version](https://img.shields.io/pypi/v/cohere-haystack.svg)](https://pypi.org/project/cohere-haystack)
 [![PyPI - Python Version](https://img.shields.io/pypi/pyversions/cohere-haystack.svg)](https://pypi.org/project/cohere-haystack)
 
+`cohere-haystack` integrates [Cohere](https://cohere.ai/) capabilities into [Haystack](https://github.com/deepset-ai/haystack) (2.x) pipelines. This package allows seamless use of Cohere's large language models (LLMs) for generating text, embedding text, and re-ranking search results. 
+
```
Comment (anakin87) [inline integrations/cohere/README.md:7]: Usually, we put all this information in https://github.com/deepset-ai/haystack-integrations.
Most of our READMEs on the present repository are instead very minimal and aimed at developers.

For consistency, I would suggest opening a PR to modify https://github.com/deepset-ai/haystack-integrations/blob/main/integrations/cohere.md.

Alternatively, let's decide together on a strategy to change this convention.

Comment (vblagoje) [reply integrations/cohere/README.md:7]: Hmm, good points indeed. Let's figure out the right thing to do as integrations are outdated, docs are outdated, an we need to figure out how to sync these and have one source of truth or a mechanism in place for an easy replication of that truth. 

I'll close this PR.

Comment (anakin87) [reply integrations/cohere/README.md:7]: Sorry!
Apart from the changes to the README, everything else (examples, changing defaults) is ok and we can merge it...

Comment (vblagoje) [reply integrations/cohere/README.md:7]: Yeah yeah, no problem, opening a new PR now with defaults and examples changes. will ping you for review @anakin87

**Pull request closed without merging**